lib/cloudregister/registerutils.py

```diff
@@ -173,6 +175,90 @@ def clear_rmt_as_scc_proxy_flag():
         pass
 
 
+# ----------------------------------------------------------------------------
+def clean_payg_extensions():
+    """Remove non free extensions files from /etc/products.d/."""
+    extensions = get_extensions()
+    products_paths = glob.glob('/etc/products.d/*.prod')
+
+    for extension in extensions:
+        if not extension.get('free'):
+            extension_path = os.path.join(
+                os.sep, 'etc', 'products.d', '{}.prod'.format(
+                    extension.get('identifier')
+                )
+            )
+            if extension_path in products_paths:
+                os.unlink(extension_path)
```

Comment:
maybe we should log here the file(s) we are removing

Reply:
yes please add a log message for the removed .prod files

Reply:
added it just right after adding the suggestion in [be49517](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/208/commits/be495177d3726f5fe4e68c424c8d0bba741218ed)